sklearn/cluster/_kmeans.py

```diff
@@ -273,7 +274,8 @@ def _tolerance(X, tol):
         "sample_weight": ["array-like", None],
         "init": [StrOptions({"k-means++", "random"}), callable, "array-like"],
         "n_init": [
-            StrOptions({"auto", "warn"}, internal={"warn"}),
+            StrOptions({"auto"}),
+            Hidden(StrOptions({"warn"})),
```

Comment:
The alternative is `StrOptions({"warn"}, hidden=True)`, but then we would need `_InstancesOf` "public", to do `InstancesOf(Criterion, hidden=True)`. Also all the other constraints will have to be "public" to be configured to hide.

With that in mind, I'm okay with the PR as is.